docker-host/main.go

```diff
@@ -0,0 +1,164 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/client"
+	"github.com/robfig/cron/v3"
+	machineryvars "github.com/uselagoon/machinery/utils/variables"
+)
+
+const EnvOverrideHost = "DOCKER_HOST"
+
+var dockerHost = machineryvars.GetEnv("DOCKER_HOST", "docker-host")
+var repositoriesToUpdate = machineryvars.GetEnv("REPOSITORIES_TO_UPDATE", "uselagoon")
+var REGISTRY = machineryvars.GetEnv("REGISTRY", "docker-registry.default.svc:5000")
+var BIP = machineryvars.GetEnv("BIP", "172.16.0.1/16")
+var REGISTRY_MIRROR = machineryvars.GetEnv("REGISTRY_MIRROR", "")
+var pruneImagesSchedule = machineryvars.GetEnv("PRUNE_SCHEDULE", "22 1 * * *")
+var removeExitedSchedule = machineryvars.GetEnv("REMOVE_EXITED_SCHEDULE", "22 */4 * * *")
+var updateImagesSchedule = machineryvars.GetEnv("UPDATE_IMAGES_SCHEDULE", "*/15 * * * *")
+var pruneImagesUntil = machineryvars.GetEnv("PRUNE_IMAGES_UNTIL", "168h")
+var danglingFilter = machineryvars.GetEnv("DANGLING_FILTER", "true")
+
+func main() {
+	cli, err := client.NewClientWithOpts(
+		client.WithHostFromEnv(),
+		client.WithAPIVersionNegotiation(),
+	)
+	if err != nil {
+		log.Fatalf("Error", err)
+	}
+	defer cli.Close()
+
+	var command = fmt.Sprintf("/usr/local/bin/dind /usr/local/bin/dockerd --host=tcp://0.0.0.0:2375 --host=unix:///var/run/docker.sock --bip=%s --storage-driver=overlay2", BIP)
```

Comment:
we should also add a var-based switch to control the log-level of the dockerd - I think by default it logs [info] - but we may want it to default to less verbose - but we can specifiy a level in var if we want?